### ISSUE TITLE
It is more or less related to issue #167 - mysql database is not getting updated with create repo / repo push

### ISSUE THREAD

**Issue by phani1kumar** [sameersbn/docker-gitlab#477]
Hi @sameersbn 

First of all thank you very  much for such a wonderful docker containers. I was using the gitlab, redis, redmine, mysql containers sucessfully until recently. My setup was:

Windows host (with Oracle Virtualbox) - Ubuntu 14.04 guest --> the guest has the docker containers for all the above mentioned. All of them have the data volumes mounted from Virtualbox Share directory.

Recently I moved my host to Ubuntu, to have a homogeneous and robust environment. After a lot of trouble I was able to set things up correctly and now have all the containers except for gitlab working. My current set-up is:

Ubuntu 14.04 host (with oracle virtualbox) - Ubuntu 14.05 guest --> the guest has been exported from windows host and imported into the linux host, thus has all the docker containers within. 
Host is sharing the volumes for the docker containers to the guest. Individual volumes are shared separately because of linux permission issues. gitlab requires the owner to be uid=1000, and gid=1000. 

Problem I am facing: 
1. When I tried to create a project from the Gitlab user interface, the creation is successful from filesystem point of view (files getting created in the shared directory). But the newly created project is not displayed in the UI. Also, there is no change in the state of mysql database files / ib_logfile0 / ib_logfile1 / ibdata1. With this I could imagine that the mysql update of the repository didn't happen. 

2. When I tried to push changes to an existing repository --> again the files are updated into the gitlab file shared directory. But the new push is not reflecting in UI and also no change to the database. 

With the above two scenarios I could imagine that there is some issue with the mysql connectivity. I checked the same from gitlab:

root@af3ee5c845e9:/# mysqladmin -h ${MYSQL_PORT_3306_TCP_ADDR} -u ${DB_USER} -p${DB_PASS} status
Uptime: 4378  Threads: 5  Questions: 630  Slow queries: 0  Opens: 105  Flush tables: 1  Open tables: 98  Queries per second avg: 0.143

The ${DB_HOST} is coming empty. Hence used the other value:MYSQL_PORT_3306_TCP_ADDR Also. The correct IP is present in the database.yml file (I've cross checked this). I've checked the logs in /var/log/gitlab/gitlab, /var/log/gitlab/gitlab-shell, /var/log/gitlab/nginx, /var/log/gitlab/supervisor but there is no error message anywhere! 

Kindly help with my situation.

**Comment by sameersbn** [comment]
Can you try the quickstart instructions to make sure it works?

**Comment by phani1kumar** [comment]
Hi @sameersbn I can see that it is somehow related to the file permissions related issue. I did the following and it started working:
1. copied the entire mount volume into the guest ubuntu server --> making it a local copy of files
2. gave 777 permissions on the folder and kept the owner to be root itself
3. ran the gitlab container, the project creation was successful

But, I don't want to have the volume directory to be on the guest machine. It should be in the host and mounted via virtualbox share folder. 
I even tried manual mount of the shared volume on the guest with 
a) "sudo mount -t vboxsf -o uid=1000,gid=1000 gitlab /mnt/Share/gitlab"  ==> these are the uid and gid of git user in the gitlab container. With this the filesystem is writable but the database is not.

b) "sudo mount -t vboxsf -o uid=999,gid=50 gitlab /mnt/Share/gitlab"  ==> these are the uid and gid of vboxadd and staff on guest ==> in some thread somewhere related to docker I found this suggestion. This didn't work, even the filesystem is not writable

**Comment by phani1kumar** [comment]
As per the gitlab documentation, the production.log should list all the sql queries fired by the system, but in my case the production.log doesn't contain any. Can you please help if there is a way I can enable this? I was searching allover in the container and not able to find any references to this.

http://doc.gitlab.com/ce/logs/logs.html

**Comment by phani1kumar** [comment]
I've also cross checked if I can connect to the database through the git user using ` sudo -u git -H mysql -h ${MYSQL_PORT_3306_TCP_ADDR} -u $DB_USER -p -D $DB_NAME`

I am able to login. I suspect some other hook before making a database insert is failing silently because of the linux file permissions. Not able to get to the root of it.

**Comment by phani1kumar** [comment]
never-mind, I solved the problem myself. The issue was with the link creation by gitlab. The gitlab was not able to create symbolic links. I've had to allow the sym link creation through 

`VBoxManage setextradata VIRTUALMACHINE_NAME VBoxInternal2/SharedFoldersEnableSymlinksCreate/SHARE_NAME 1` 
VIRTUALMACHINE_NAME should be the name of the guest, and SHRE_NAME shall be the absolute path of the share directory from host. After the setting is changed, the host need to be rebooted for it to take effect. 

Also, as I've migrated from one server to the other, I've had to re-create the symlinks manually using
`find . -name hooks -exec sh -c 'ln -snf /home/git/gitlab-shell/hooks $0' {} \;`

Also, the gitlab installation can be validated through the following steps:

```
sudo -i -u git
cd gitlab
bundle exec rake gitlab:env:info RAILS_ENV=production
bundle exec rake gitlab:check RAILS_ENV=production
```
Extracted from https://github.com/gitlabhq/gitlabhq/blob/master/doc/development/architecture.md
